contentctl/objects/abstract_security_content_objects/detection_abstract.py

```diff
@@ -60,6 +60,34 @@ class Detection_Abstract(SecurityContentObject):
     class Config:
         use_enum_values = True
 
+
+    # A list of groups of tests, relying on the same data
+    test_groups: Union[list[TestGroup], None] = None
```

Comment:
I see you migrated this code from Detection

Can you explain the intention behind having separate classes here (detection and the abstract)? I'm fine w/ this code living in either class, but I chose Detection bc I thought the spirit of the abstract class was to be a more faithful representation of the model as it exists in YAML, and since the `test_groups` attribute is an inferred field, I thought maybe it made sense to live elsewhere.

So tl;dr I'm totally fine with this, and no change required, just curious, and wondering if we don't have a need for two separate classes, perhaps we could collapse them into one in the future

Reply:
Great question!  Many contentctl users want to change content somehow (and detections are the content they are most-often interested in changing).  These changes could be, but are not limited to:
1.  Adding new fields (and possibly building validators for those fields)
2.  Setting more or less strict validations on fields
3.  Setting values for fields already defined in that object (but which may be missing from their YMLs)
4. Disabling some of the validations we do.

However, if they change Detection_Abstract and WE change Detection_Abstract, it has a high probability of creating painful merge conflict issues.  As such, the plan was to retain control of Detection_Abstract and ship all our out-of-the-box functionality in that file.  Then if customers want their own functionality, they can add it to Detection.py.
This way, pulling the latest contentctl updates should be painless and without merge conflicts.  We are hoping to do this with all types, but have started with Detection since it is the one that receives user changes most often.